core/unpack/unpacker_test.go

```diff
@@ -0,0 +1,99 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package unpack
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/image-spec/identity"
+)
+
+func generateDiffIDs(num int) []digest.Digest {
+	// sample diffIDs from golang@sha256:ff787165114376423b784ea19442b4a7988c58789f3aa97560f57f0aea87460e
+	samples := []digest.Digest{
+		"sha256:6abe10f2f60150ae9768e117986e4f8af5546137221553228eb5d21066f596b4",
+		"sha256:25879f85bbb0206e6870e7073a576acd49be6e8c3a15ba4b443166f92b041327",
+		"sha256:045d8b74bf0ddedb671bd1ba1ab114e3a3e7b86dc5c7e18f9b7edda74193174d",
+		"sha256:eb8d59a4555330682869876c48ba1d5d4339b3f66e05eddebff14bac883284ac",
+		"sha256:9538b1dfedad41772b0870d6b4fe4df87ad5123031e9f669c9995c2d1c035e44",
+		"sha256:ab8ccb753029aea342285516d925abebc018baaf5ab85a94c11784980a721974",
+		"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef",
+	}
+	diffIDs := make([]digest.Digest, 0, num)
+	for {
+		diffIDs = append(diffIDs, samples...)
+		if len(diffIDs) >= num {
+			break
+		}
```

Comment:
What if num > len(samples)?

Reply:
Probably you should just use `crypt/rand` to generate random IDs

Reply:
sounds good, updated